### ISSUE TITLE
i get some warnings on build

### ISSUE THREAD

**Issue by pishguy** [sgl0v/SCrypto#9]
hi, how can i resolve this problems?

```
mahdipishguy@mahdis-Mac Runner % flutter run       
Changing current working directory to: /Users/mahdipishguy/Documents/projects/unlimited_power_pro
Launching lib/main.dart on iPhone 8 in debug mode...
 
Running Xcode build...                                                  
                                                   
Xcode build done.                                           24.3s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:107:111: error: using '!' is not allowed here; perhaps '?' was intended?
            typealias Function = (_ data: UnsafeRawPointer, _ len: CC_LONG, _ md: UnsafeMutablePointer<UInt8>) -> UnsafeMutablePointer<UInt8>!
                                                                                                                  ^                          ~
                                                                                                                                             ?
    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:188:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func MD2() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:198:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func MD4() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:208:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func MD5() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:217:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func SHA1() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:226:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func SHA224() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:235:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func SHA256() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:244:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func SHA384() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:253:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func SHA512() -> Self {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:330:5: warning: 'public' modifier is redundant for static method declared in a public extension
        public static func random(_ length : Int) throws -> Data {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:432:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func hmac(_ algorithm: HMAC.Algorithm, key: Data) -> Data {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:453:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func hmac(_ algorithm: HMAC.Algorithm, key: String) -> String {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:628:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func encrypt(_ algorithm: Cipher.Algorithm, options: Cipher.Options, key: Data, iv: Data? = nil) throws -> Data {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:646:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func decrypt(_ algorithm: Cipher.Algorithm, options: Cipher.Options, key: Data, iv: Data? = nil) throws -> Data {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:745:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func derivedKey(_ salt: Data, pseudoRandomAlgorithm: PBKDF.PseudoRandomAlgorithm, rounds: UInt32, derivedKeyLength: Int) throws -> Data {
        ^~~~~~~

    /Users/mahdipishguy/Documents/projects/unlimited_power_pro/ios/Pods/SCrypto/Source/SCrypto.swift:760:5: warning: 'public' modifier is redundant for instance method declared in a public extension
        public func calibrate(_ saltLength: Int, pseudoRandomAlgorithm: PBKDF.PseudoRandomAlgorithm, derivedKeyLength: Int, msec : UInt32) -> UInt {
        ^~~~~~~

    note: Using new build system
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 8.
mahdipishguy@mahdis-Mac Runner % 
```

**Comment by acoutts** [comment]
I'm getting the same error.

**Comment by sgl0v** [comment]
Should be fixed in version 4.0.0
